subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/InjectTransformParameters.java

```diff
@@ -17,20 +17,23 @@
 package org.gradle.api.artifacts.transform;
 
 import org.gradle.api.Incubating;
+import org.gradle.api.reflect.InjectionPointQualifier;
 
+import java.lang.annotation.Documented;
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
 /**
- * Attached to an artifact transform parameter type to declare the corresponding {@link TransformAction} implementation to use.
+ * Attached to a property that should receive the <em>parameter object</em> of the artifact transform.
  *
  * @since 5.3
  */
 @Incubating
 @Retention(RetentionPolicy.RUNTIME)
-@Target({ElementType.TYPE})
-public @interface AssociatedTransformAction {
-    Class<? extends TransformAction> value();
+@Target({ElementType.METHOD})
+@Documented
+@InjectionPointQualifier
+public @interface InjectTransformParameters {
```

Comment:
I think this should be in an internal package. If we want to keep this public API, we should say that it is not to be used by the user.

Reply:
I want to remove the annotation, only Adam's changes have not been merged, yet.

Reply:
I removed the annotation.